File: gdscript/src/main/java/gdscript/psi/GdConstDeclTl.java
package gdscript.psi;

import com.intellij.navigation.ItemPresentation;
import com.intellij.psi.StubBasedPsiElement;
import gdscript.index.stub.GdConstDeclStub;
import gdscript.psi.types.GdDocumented;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public interface GdConstDeclTl extends GdTopLevelDecl, StubBasedPsiElement<GdConstDeclStub>, GdDocumented {

  @Nullable
  GdAssignTyped getAssignTyped();

  @Nullable
  GdEndStmt getEndStmt();

  @Nullable
  GdExpr getExpr();

  @Nullable
  GdTyped getTyped();

  @Nullable
  GdVarNmi getVarNmi();

  @NotNull
  String getName();

  @NotNull
  String getReturnType();

  @NotNull
  ItemPresentation getPresentation();

}

File: gdscript/src/main/kotlin/gdscript/psi/GdConstDeclTl.kt
package gdscript.psi

import com.intellij.navigation.ItemPresentation
import com.intellij.psi.StubBasedPsiElement
import gdscript.index.stub.GdConstDeclStub
import gdscript.psi.types.GdDocumented

interface GdConstDeclTl : GdTopLevelDecl, StubBasedPsiElement<GdConstDeclStub>, GdDocumented {
    val assignTyped: GdAssignTyped?

    val endStmt: GdEndStmt?

    val expr: GdExpr?

    val typed: GdTyped?

    val varNmi: GdVarNmi?

    fun getName(): String

    val returnType: String

    fun getPresentation(): ItemPresentation
}
